samples/parselib/src/main/cpp/wav/WavStreamReader.cpp

```diff
@@ -39,7 +39,7 @@ WavStreamReader::WavStreamReader(InputStream *stream) {
 
     mAudioDataStartPos = -1;
 
-    mChunkMap = new std::map<RiffID, WavChunkHeader *>();
+    mChunkMap = new std::map<RiffID, std::shared_ptr<WavChunkHeader>>();

```

Comment:
Where does this get deleted?

Reply:
More to the point, why bother allocating it dynamically in any case?